### PR TITLE
Emit warning-level logs when truncating messages in notifications

### PR DESCRIPTION
**Pull request by alexweav**
Some integrations define maximum lengths of messages that they accept. We usually respect these, by truncating the user's message for them.

Since we are actively modifying what the user asked us to send, let's log this event at warning level. That way, operators can more easily see when their message is too long, and know that we changed it for them.

Fixes https://github.com/prometheus/alertmanager/issues/3134

We no longer log the full-blown message per https://github.com/prometheus/alertmanager/issues/3134#issuecomment-1315343535.

### CODE REVIEW

File: notify/opsgenie/opsgenie.go [hunk 1]
```diff
@@ -171,10 +174,9 @@ func (n *Notifier) createRequests(ctx context.Context, as ...*types.Alert) ([]*h
 		}
 		requests = append(requests, req.WithContext(ctx))
 	default:
-		// https://docs.opsgenie.com/docs/alert-api - 130 characters meaning runes.
-		message, truncated := notify.TruncateInRunes(tmpl(n.conf.Message), 130)
+		message, truncated := notify.TruncateInRunes(tmpl(n.conf.Message), maxMessageLenRunes)
 		if truncated {
-			level.Debug(n.logger).Log("msg", "truncated message", "truncated_message", message, "alert", key)
+			level.Debug(n.logger).Log("msg", "Truncated message", "alert", key, "runes", maxMessageLenRunes)
```
Comment (alexweav) [inline notify/opsgenie/opsgenie.go]: Standardized the capitalization of these to use a capital at the beginning. Please let me know if you'd rather me go the other way.

File: notify/pagerduty/pagerduty.go [hunk 2]
```diff
@@ -36,7 +36,13 @@ import (
 	"github.com/prometheus/alertmanager/types"
 )
 
-const maxEventSize int = 512000
+const (
+	maxEventSize int = 512000
+	// https://developer.pagerduty.com/docs/ZG9jOjExMDI5NTc4-send-a-v1-event - 1024 characters or runes.
```
Comment (alexweav) [inline notify/pagerduty/pagerduty.go:41]: Fixed link and number.

**Review by simonpasquier — APPROVED** [review]
lgtm

leaving the final word to @gotjosh

**Comment by alexweav** [comment]
hey @gotjosh, any final comments here?

File: notify/opsgenie/opsgenie.go [hunk 3]
```diff
@@ -171,10 +174,9 @@ func (n *Notifier) createRequests(ctx context.Context, as ...*types.Alert) ([]*h
 		}
 		requests = append(requests, req.WithContext(ctx))
 	default:
-		// https://docs.opsgenie.com/docs/alert-api - 130 characters meaning runes.
-		message, truncated := notify.TruncateInRunes(tmpl(n.conf.Message), 130)
+		message, truncated := notify.TruncateInRunes(tmpl(n.conf.Message), maxMessageLenRunes)
 		if truncated {
-			level.Debug(n.logger).Log("msg", "truncated message", "truncated_message", message, "alert", key)
+			level.Warn(n.logger).Log("msg", "Truncated message", "alert", key, "runes", maxMessageLenRunes)
```
Comment (gotjosh) [inline notify/opsgenie/opsgenie.go:179]: I think the "runes" key should be "max_runes", WDYT? off the bat, I wouldn't know if this is the current number of runes of the message that got truncated or the max length supported by the platform.

**Review by gotjosh — APPROVED** [review]
Overall, LGTM but I have one small nit.

File: notify/opsgenie/opsgenie.go (hunk 3 above)
Comment (alexweav) [reply notify/opsgenie/opsgenie.go]: Done, good idea

**Pull request merged**